templates/template_helpers.go

```diff
@@ -167,26 +173,109 @@ func extractSnippetName(line string) (string, bool) {
+	return func(value interface{}) (float64, error) {
+		valueAsFloat, err := toFloat64(value)
 		if err != nil {
-			return "", errors.WithStackTrace(err)
+			return 0, errors.WithStackTrace(err)
 		}
-		return strconv.Itoa(int(f(valueAsFloat))), nil
+		return f(valueAsFloat), nil
 	}
 }
 
-// Wrap a function that uses float64 as input and int as output so it can take a string as input and return a string as
+// Wrap a function that uses float64 as input and int as output so it can take any number as input and return an int as
 // output
-func wrapFloatToIntFunction(f func(float64) int) func(string) (string, error) {
-	return func(valueAsString string) (string, error) {
-		valueAsFloat, err := strconv.ParseFloat(valueAsString, 64)
+func wrapFloatToIntFunction(f func(float64) int) func(interface{}) (int, error) {
+	return func(value interface{}) (int, error) {
+		valueAsFloat, err := toFloat64(value)
+		if err != nil {
+			return 0, errors.WithStackTrace(err)
+		}
+		return f(valueAsFloat), nil
+	}
+}
+
+// Wrap a function that takes two ints as input and returns an int as output so it can take any kind of number as input
+// and return an int as output
+func wrapIntIntToIntFunction(f func(int, int) int) func(interface{}, interface{}) (int, error) {
+	return func(arg1 interface{}, arg2 interface{}) (int, error) {
+		arg1AsInt, err := toInt(arg1)
+		if err != nil {
+			return 0, errors.WithStackTrace(err)
+		}
+		arg2AsInt, err := toInt(arg2)
 		if err != nil {
-			return "", errors.WithStackTrace(err)
+			return 0, errors.WithStackTrace(err)
 		}
-		return strconv.Itoa(f(valueAsFloat)), nil
+		return f(arg1AsInt, arg2AsInt), nil
+
+	}
+}
+
+// Wrap a function that takes two float64's as input, performs arithmetic on them, and returns another float64 as a
+// function that can take two values of any number kind as input and return a float64 as output
+func wrapFloatFloatToFloatFunction(f func(arg1 float64, arg2 float64) float64) func(interface{}, interface{}) (float64, error) {
+	return func(arg1 interface{}, arg2 interface{}) (float64, error) {
+		arg1AsFloat, err := toFloat64(arg1)
+		if err != nil {
+			return 0, errors.WithStackTrace(err)
+		}
+		arg2AsFloat, err := toFloat64(arg2)
+		if err != nil {
+			return 0, errors.WithStackTrace(err)
+		}
+		return f(arg1AsFloat, arg2AsFloat), nil
+
+	}
+}
+
+// Convert the given value to a float64. Does a proper conversion if the underlying type is a number. For all other
+// types, we first convert to a string, and then try to parse the result as a float64.
+func toFloat64(value interface{}) (float64, error) {
+	// Because Go is a shitty language, we have to call out each of the numeric types separately, even though the
+	// behavior for almost all of them is identical. If we tried to do a case statement with multiple clauses
+	// (separated by comma), then the variable v would be of type interface{} and we could not use float64(..) to
+	// convert it.
```

Comment:
I was getting worried you wouldn't complain about Go in this PR, but my fears are now allayed.